lib/services/access_request.go

```diff
@@ -1670,11 +1670,17 @@ func (m *RequestValidator) getRequestableRoles(ctx context.Context, identity tls
 	// Filter out resources the user requested but doesn't have access to.
 	filteredResources := make([]types.ResourceWithLabels, 0, len(resources))
 	for _, resource := range resources {
-		if err := accessChecker.CheckAccess(resource, AccessState{MFAVerified: true}); err == nil {
-			filteredResources = append(filteredResources, resource)
+		switch resource.GetKind() {
+		case types.KindSAMLIdPServiceProvider:
+			if err = accessChecker.CheckAccessToSAMLIdP(resource, nil /*AuthPreference*/, AccessState{MFAVerified: true}); err != nil {
+				filteredResources = append(filteredResources, resource)
+			}
```

Comment:
At first glance this looks like sec bug where the rescues is added when resource is added when the err != nil -> Access Denied 

> // Filter out resources the user requested but doesn't have access to.

Reply:
This was addressed in https://github.com/gravitational/teleport/pull/61204/commits/91179055625561d87c09fe35614aaa95c04bc9b1